ext/pdo/pdo_dbh.c

```diff
@@ -498,42 +497,47 @@ PHP_METHOD(PDO, prepare)
 		Z_PARAM_ARRAY(options)
 	ZEND_PARSE_PARAMETERS_END();
 
-	PDO_DBH_CLEAR_ERR();
 	PDO_CONSTRUCT_CHECK;
 
-	if (ZEND_NUM_ARGS() > 1 && (opt = zend_hash_index_find(Z_ARRVAL_P(options), PDO_ATTR_STATEMENT_CLASS)) != NULL) {
-		if (Z_TYPE_P(opt) != IS_ARRAY || (item = zend_hash_index_find(Z_ARRVAL_P(opt), 0)) == NULL
-			|| Z_TYPE_P(item) != IS_STRING
-			|| (pce = zend_lookup_class(Z_STR_P(item))) == NULL
-		) {
-			pdo_raise_impl_error(dbh, NULL, "HY000",
-				"PDO::ATTR_STATEMENT_CLASS requires format array(classname, array(ctor_args)); "
-				"the classname must be a string specifying an existing class"
-				);
-			PDO_HANDLE_DBH_ERR();
-			RETURN_FALSE;
+	if (statement_len == 0) {
+		zend_argument_value_error(1, "cannot be empty");
+		RETURN_THROWS();
+	}
+
+	PDO_DBH_CLEAR_ERR();
+
+	if (options && (value = zend_hash_index_find(Z_ARRVAL_P(options), PDO_ATTR_STATEMENT_CLASS)) != NULL) {
+		if (Z_TYPE_P(value) != IS_ARRAY) {
+			zend_type_error("PDO::ATTR_STATEMENT_CLASS's value must be of type array, %s given",
```

Comment:
```suggestion
			zend_type_error("PDO::ATTR_STATEMENT_CLASS value must be of type array, %s given",
```
The `'s` looks pretty awkward to me here. It's okay to leave it off.